### PR TITLE
Create menu

### PR DESCRIPTION
**Pull request by taniarezhytska**

### CODE REVIEW

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿
+namespace TestFramework.PageComponents.Owner.CreateMenuPageComponents
+{
+    public class CreateListMenuAndComponent
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs:4]: The name of the class must end with ...PageComponent - this is our template. Please rename all your classes for page components.

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+            public CreateListMenuAndComponent(IWebDriver driver)
+            {
+                this.driver = driver;
+            }
+
+            private IWebElement _finishButton => driver.FindElement(By.XPath("//p[contains(text(), 'Try out')]/parent::div/following-sibling::div[2]"));
+            private IWebElement _backButton => driver.FindElement(By.XPath("//p[contains(text(), 'Try out')]/parent::div/following-sibling::div[1]"));
+
+            public CreateListMenuAndComponent ClickThreeDotsButton()
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs:16]: The name is ThreeDotsButton, but you click on finish button inside the method. Please make changes.

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    {
+            IWebDriver driver { get; }
+
+            public CreateListMenuAndComponent(IWebDriver driver)
+            {
+                this.driver = driver;
+            }
+
+            private IWebElement _finishButton => driver.FindElement(By.XPath("//p[contains(text(), 'Try out')]/parent::div/following-sibling::div[2]"));
+            private IWebElement _backButton => driver.FindElement(By.XPath("//p[contains(text(), 'Try out')]/parent::div/following-sibling::div[1]"));
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs:14]: I'm afraid you won't find these buttons until you make 2 steps before this.
image

Comment (LoveMyDaughter) [reply TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs:14]: Back and Finish buttons are visible only on the third step.

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuComponent.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+﻿using TestFramework.Pages.Owner;
+
+namespace TestFramework.PageComponents.Owner.CreateMenuPageComponents
+{
+    public class CreateListMenuComponent
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuComponent.cs:5]: I can't find this component in the browser. Please give me a screenshot.

File: TAF_EasyRest/TestFramework/Pages/Owner/CreateMenuPage.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        public NavigationMenuPageComponent NavigationMenuPageComponent { get; }
+        public ManageRestaurantPageComponent ManageRestaurantPageComponent { get; }
+
+        public CreateMenuPage(IWebDriver driver) : base(driver)
+        {
+            NavigationMenuPageComponent = new NavigationMenuPageComponent(driver);
+            ManageRestaurantPageComponent = new ManageRestaurantPageComponent(driver);
+        }
+        private IWebElement _menuNameField => driver.FindElement(By.Name("Menu Name"));
+        private IWebElement _listMenuButton => driver.FindElement(By.XPath("//span[text() = 'List menu']"));
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/Pages/Owner/CreateMenuPage.cs:18]: Would be better to mention that this is a radio button: _listMenuRaioButton

File: TAF_EasyRest/TestFramework/Pages/Owner/CreateMenuPage.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        private IWebElement _imageMenuButton => driver.FindElement(By.XPath("//span[text() = 'Image menu']"));
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/Pages/Owner/CreateMenuPage.cs:19]: Would be better to mention that this is a radio button: _imageMenuRaioButton

File: TAF_EasyRest/TestFramework/Pages/Owner/CreateMenuPage.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        private IWebElement _nextButton => driver.FindElement(By.XPath("//legend[contains(text(),'Choose menu type:')]/parent::div/parent::div/following-sibling::div[2]"));
+
+        public CreateMenuPage ClickMenuField()
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/Pages/Owner/CreateMenuPage.cs:22]: You actually click on _menuNameField - so the method name must be ClickMenuNameField.

**Review by LoveMyDaughter — COMMENTED** [review]
Please make changes.

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs (hunk 3 above)
Comment (taniarezhytska) [reply TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs:14]: so I moved them to another component

Comment (taniarezhytska) [reply TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuAndComponent.cs:14]: this component appears when we click "next" on the previously page

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuEndPageComponent.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
+﻿
+namespace TestFramework.PageComponents.Owner.CreateMenuPageComponents
+{
+    public class CreateListMenuEndPageComponent
+    {
+            IWebDriver driver { get; }
```
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuEndPageComponent.cs:6]: please specify the access modifier explicitly

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuPageComponent.cs [hunk 9]
```diff
@@ -0,0 +1,37 @@
+﻿using TestFramework.Pages.Owner;
+
+namespace TestFramework.PageComponents.Owner.CreateMenuPageComponents
+{
+    public class CreateListMenuPageComponent 
+    {
+        IWebDriver driver { get; }
```
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuPageComponent.cs:7]: please specify the access modifier explicitly

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuEndPageComponent.cs [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+            {
+                _finishButton.Click();
+                return this;
+            }
+            public CreateListMenuPageComponent ClickBackButton()
+            {
+                _backButton.Click();
+                return new CreateListMenuPageComponent(driver);
+            }
+
```
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuEndPageComponent.cs:26]: please remove empty lines

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuPageComponent.cs [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+            return this;
+        }
+        public CreateListMenuEndPageComponent ClickNextButton()
+        {
+            _nextButton.Click();
+            return new CreateListMenuEndPageComponent(driver);
+        }
+
+        public CreateMenuPage ClickBackButton(
+            )
```
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuPageComponent.cs:30]: please format the line

**Review by dmatsev — CHANGES_REQUESTED** [review]
Please resolve the comments.

File: TAF_EasyRest/TestFramework/Pages/Owner/CreateMenuPage.cs [hunk 12]
```diff
@@ -0,0 +1,49 @@
+﻿using TestFramework.PageComponents;
+using TestFramework.PageComponents.NavigationMenuComponents;
+using TestFramework.PageComponents.Owner.CreateMenuPageComponents;
+
+namespace TestFramework.Pages.Owner
+{
+    public class CreateMenuPage : BasePage
+    {
+        public NavigationMenuPageComponent NavigationMenuPageComponent { get; }
+        public ManageRestaurantPageComponent ManageRestaurantPageComponent { get; }
```
Comment (andriifromlviv) [inline TAF_EasyRest/TestFramework/Pages/Owner/CreateMenuPage.cs:10]: you probably need to add " UserMenuHeaderButtonPageComponent" here as we use this page for user who has signed in already

**Review by andriifromlviv — CHANGES_REQUESTED** [review]

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuEndPageComponent.cs [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    {
+            private IWebDriver driver { get; }
+            public CreateListMenuEndPageComponent(IWebDriver driver)
+            {
+                this.driver = driver;
+            }
+
+            private IWebElement _finishButton => driver.FindElement(By.XPath("//p[contains(text(), 'Try out')]/parent::div/following-sibling::div[2]"));
+            private IWebElement _backButton => driver.FindElement(By.XPath("//p[contains(text(), 'Try out')]/parent::div/following-sibling::div[1]"));
+            public CreateListMenuEndPageComponent ClickFinishButton()
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuEndPageComponent.cs:14]: At this step, you get a new component on the leftside menu (concretely a new menu item). You should define this somehow.

File: TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuPageComponent.cs [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+        public CreateListMenuPageComponent(IWebDriver driver)
+        {
+            this.driver = driver;
+        }
+  
+        private IWebElement _threeDotsButton => driver.FindElement(By.XPath("//button[@aria-label='Action']"));
+        private IWebElement _nextButton => driver.FindElement(By.XPath("//button[@aria-label='Action']/parent::div/parent::div/parent::div/parent::div/following-sibling::div[2]"));
+        private IWebElement _backButton => driver.FindElement(By.XPath("//button[@aria-label='Action']/parent::div/parent::div/parent::div/parent::div/following-sibling::div[1]"));
+
+        public CreateListMenuPageComponent ClickThreeDotsButton()
```
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/PageComponents/Owner/CreateMenuPageComponents/CreateListMenuPageComponent.cs:17]: Don't you need to define a new appeared element here?
image

**Review by LoveMyDaughter — CHANGES_REQUESTED** [review]
Please add the UserButton as Andriy mentioned previously. Class CreateMenuPage line 11.

**Review by dmatsev — APPROVED** [review]

**Review by LoveMyDaughter — APPROVED** [review]

**Review by sofiaso — APPROVED** [review]

**Pull request merged**